ui/js/controllers/bugfiler.js

```diff
@@ -0,0 +1,224 @@
+            }
+
+            // Look up product suggestions via Bugzilla's api
+            var productSearch = $scope.productSearch;
+
+            if(productSearch) {
+                $http.get("https://bugzilla.mozilla.org/rest/prod_comp_search/" + productSearch + "?limit=5").then(function(request) {
+                    var data = request.data;
+                    $scope.suggestedProducts = [];
+                    for(var i = 0; i < data.products.length;i++) {
+                        if(data.products[i].product && data.products[i].component) {
+                            $scope.suggestedProducts.push(data.products[i].product + " :: " + data.products[i].component);
+                        }
+                    }
+                });
+            }
+        };
+
+        /*
+         *  This is called once intermittent.html's ng-repeat is finished to select the first product listed
+         */
+        $scope.focusProduct = function() {
+            $("#suggestedProducts").children(":first").children(":first").prop("checked", true);
+        };
+
+        /*
+         *  Same as clicking outside of the modal, but with a nice button-clicking feel...
+         */
+        $scope.cancelFiler = function() {
+            $uibModalInstance.dismiss('cancel');
+        };
+
+        /*
+         *  Actually send the gathered information to bugzilla.
+         */
+        $scope.submitFiler = function() {
+            var bugzillaRoot = "https://bugzilla-dev.allizom.org/"; // (prod is "https://bugzilla.mozilla.org/");
+            var summarystring = $scope.modalSummary;
+            var productString = "";
+            var componentString = "";
+            var isProductSelected = false;
+
+            $(':input','#modalForm').attr("disabled",true);
+
+            var productRadios = $("#modalForm input[name='productGroup'");
+            if(productRadios && productRadios.length) {
+                for(var i=0, len=productRadios.length; i<len; i++) {
+                    if(productRadios[i].checked) {
+                        productString += productRadios[i].value.split(" :: ")[0];
+                        componentString += productRadios[i].value.split(" :: ")[1];
+                        isProductSelected = true;
+                        break;
+                    }
+                }
+            } else {
+                if(productRadios && productRadios.checked) {
+                    productString += productRadios.value.split(" :: ")[0];
+                    componentString += productRadios.value.split(" :: ")[1];
+                    isProductSelected = true;
+                }
+            }
+
+            if(!isProductSelected) {
+                alert("Please select (or search and select) a product/component pair to continue");
+                return;
+            }
+            var logstrings = "";
+            var logcheckboxes = document.getElementById("modalLogLinkCheckboxes").getElementsByTagName("input");
```

Comment:
There are a few ways to approach this situation with the log links, but here's one:

Outside this function, declare a $scope var like this:

    $scope.checkedLogLinks = {
        parsedLog: $scope.parsedLog,
        fullLog: $scope.fullLog,
        reftest: $scope.reftest
    };

Then in the html for each ``<input>`` remove the ``checked=true`` part and add  this:

    ng-model="checkedLogLinks.parsedLog"
    ng-true-value="'{{parsedLog}}'"/>

changing as appropriate for each check.  This will keep track of the desired log links in the $scope variable of ``checkedLogLinks``.  It'll save the actual text of the links in there for easy retrieval.   Then, replace the next for loop with something like this:

    var logstrings = _.reduce($scope.checkedLogLinks, function(result, link) {
        if(link) {
            result = result + link + "\n\n";
        }
        return result;
    }, "");

I know this doesn't really make the code much shorter or anything, but it's easier to read and follow, I think.  :)